csm/core/controllers/rgw/s3/users.py

```diff
@@ -79,3 +111,58 @@ async def put(self):
         with self._guard_service():
             response = await self._service.create_user(**user_body)
             return CsmResponse(response)
+
+@CsmView._app_routes.view("/api/v2/s3/iam/users/keys")
+class S3IAMUserKeyView(S3BaseView):
+    """
+    S3 IAM User List View for REST API implementation.
+
+    PUT: Add/Create access key
+    """
+
+    def __init__(self, request):
+        """S3 IAM User List View Init."""
+        super().__init__(request)
+        self._service = self.request.app[const.RGW_S3_IAM_USERS_SERVICE]
+
+    @CsmAuth.permissions({Resource.S3_IAM_USERS: {Action.UPDATE}})
```

Comment:
What should be the appropriate Action ? update or create ? 
if update, then u might need to give update access rights to 's3_iam_users' resource.